build.gradle.kts

```diff
@@ -0,0 +1,199 @@
+import org.gradle.api.tasks.bundling.Jar
+import kotlin.concurrent.thread
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.io.InputStream
+import org.jetbrains.kotlin.gradle.dsl.Coroutines
+
+plugins {
+    `java-library`
+    `maven-publish`
+    kotlin("jvm") version "1.2.61"
+    id("net.saliman.properties") version "1.4.6"
+}
+
+fun gitBranch() =
+        if (System.getenv("TRAVIS_BRANCH") != null) System.getenv("TRAVIS_BRANCH")!!
+        else {
+            val r = "git rev-parse --abbrev-ref HEAD".execute(rootDir.absolutePath)
+            r.lineSequence().last()
+        }
+
+val branch = gitBranch()
+logger.info("On branch $branch")
+
+group = "be.bluexin"
+version = "${if (branch == "master") "" else "$branch-"}${prop("version_number")}.${prop("build_number")}"
+
+repositories {
+    jcenter()
+}
+
+configurations {
+    arrayOf(
+            "shade",
+            "shadeInPlace"
+    ).forEach {
+        create(it) {
+            get("compileOnly").extendsFrom(this)
+            get("testCompileOnly").extendsFrom(this)
+        }
+    }
+}
+
+dependencies {
+    api(kotlin("stdlib-jdk8"))
+    api("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+    api("org.jetbrains.kotlinx:kotlinx-coroutines-core:0.19.3")
+    api("org.jetbrains.kotlinx:kotlinx-coroutines-jdk8:0.19.3")
+    api("org.jetbrains.kotlinx:kotlinx-coroutines-nio:0.19.3")
+
+    implementation("org.slf4j:slf4j-api:1.7.25")
+    implementation("io.github.microutils:kotlin-logging:1.6.10")
+    implementation("org.slf4j:slf4j-simple:1.7.25")
```

Comment:
this has to be `runtimeOnly`, otherwise it will leak into dependent projects

Reply:
Well from how I understood it (counterpart of `api`) it should be marked implementation so dependents will have it on their runtime path but not on their compile path?
Or should this not even get to consumers run path?

Reply:
It shouldn't even get into the consumers runtime classpath, only in your runtime, otherwise you force them to use a specific slf4j implementation, rendering slf4j useless. This is basically only for your testing.

Reply:
Right, I forgot about that.
I should leave kotlin-logging as is though right?

Reply:
yes, since it is required at compile time